modules/govuk_containers/manifests/terraboard.pp

```diff
@@ -0,0 +1,130 @@
+# == Class: govuk_containers::terraboard
+#
+# Install and run Terraboard in a Docker container with PostgreSQL backend
+# and OAuth2 Proxy for authentication, also in Docker containers. All containers
+# are connected to the `terranet` bridge network.
+#
+# The PostgreSQL database is used to cache Terraform states to reduce the amount
+# of traffic.
+#
+# === Parameters
+#
+# [*aws_region*]
+#   The AWS region to fetch Terraform state from.
+#
+# [*aws_access_key_id*]
+#   The AWS access key ID to use when fetching Terraform state.
+#
+# [*aws_secret_access_key*]
+#   The AWS secret access key to use when fetching Terraform state.
+#
+# [*aws_bucket*]
+#   The AWS bucket to fetch the Terraform state from.
+#
+# [*db_password*]
+#   The database password for the PostgreSQL Docker instance used by Terraboard.
+#
+# [*github_oauth_client_id*]
+#   The OAuth client ID provided by GitHub for authentication.
+#
+# [*github_oauth_client_secret*]
+#   The OAuth client secret provided by GitHub for authentication.
+#
+# [*oauth2_proxy_base_url*]
+#   The base URL of the OAuth2 Proxy installation.
+#
+# [*oauth2_proxy_cookie_secret*]
+#   A Base64-encoded secret used to encrypt the OAuth2 Proxy cookie.
+#
+class govuk_containers::terraboard(
+  $aws_region = 'eu-west-1',
+  $aws_access_key_id = undef,
+  $aws_secret_access_key = undef,
+  $aws_bucket = undef,
+  $db_password = undef,
+  $github_oauth_client_id = undef,
+  $github_oauth_client_secret = undef,
+  $oauth2_proxy_base_url = undef,
+  $oauth2_proxy_cookie_secret = undef,
+) {
+
+  file { '/opt/terraboard':
```

Comment:
Might be nice to combine this and the one below with something like:

```puppet
file { ['/opt/terraboard', '/opt/terraboard/conf']:
  ensure => directory,
  ...
}
```

Reply:
TIL!